src/chainstate/stacks/block.rs

```diff
@@ -186,20 +186,16 @@ impl StacksBlockHeader {
     /// This is the "block hash" used for extending the state index root.
     /// This method is necessary because the index root must be globally unique (but, the same stacks
     /// block header can show up multiple times on different burn chain forks).
-    pub fn make_index_block_hash(burn_block_hash: &BurnchainHeaderHash, block_hash: &BlockHeaderHash) -> BlockHeaderHash {
-        let mut hash_bytes = vec![];
-        
-        hash_bytes.extend_from_slice(&mut block_hash.as_bytes().clone());
-        hash_bytes.extend_from_slice(&mut burn_block_hash.as_bytes().clone());
-        
-        let h = Sha512Trunc256Sum::from_data(&hash_bytes);
-        let mut b = [0u8; 32];
-        b.copy_from_slice(h.as_bytes());
+    pub fn make_index_block_hash(burn_block_hash: &BurnchainHeaderHash, block_hash: &BlockHeaderHash) -> StacksBlockId {
```

Comment:
I'm trying to understand how this method is going to change once `StacksBlockId` is derived from whether or not the block it represents occurs within a reward cycle with or without an anchored block.  Is this method signature going to change to include a third "`has_pox_anchor_block`" bit argument, or is the `BurnchainHeaderHash` going to be altered so it commits to the `has_pox_anchor_block` bit already?

Reply:
There will be a new type `BurnBlockId` that this will accept instead (and the burndb will use for it's MARF identifier). The `BurnBlockId` will itself have a `make` function that accepts `BurnchainHeaderHash` and `PoxForkId`.